util/pull/pull.go

```diff
@@ -144,6 +159,12 @@ func (p *Puller) Pull(ctx context.Context) (*Pulled, error) {
 	}
 	stopProgress()
 
+	if isLegacyConvertible {
+		if p.desc, err = docker.ConvertManifest(ctx, p.ContentStore, p.desc); err != nil {
```

Comment:
Why is the conversion needed here? Wouldn't it be better to leave the manifest as is and make sure `images.Config()` can handle the invalid media-type. buildkit should never reuse this mediatype anyway when creating new images as it only tracks layers.